pandas/tseries/period.py

```diff
@@ -271,10 +271,13 @@ def _from_arraylike(cls, data, freq, tz):
 
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        if not getattr(values, 'dtype', None):
+
+        if not com.is_integer_dtype(values):
```

Comment:
no that is fine; if its already int64 it will just return (which is what you want), and coerce int32 -> int64 (which is also what we want)